src/Cms/Page.php

```diff
@@ -1029,11 +1029,19 @@ public function render(array $data = [], $contentType = 'html'): string
 
         // try to get the page from cache
         if (empty($data) === true && $this->isCacheable() === true) {
-            $cache    = $kirby->cache('pages');
-            $cacheId  = $this->cacheId($contentType);
-            $result   = $cache->get($cacheId);
-            $html     = $result['html'] ?? null;
-            $response = $result['response'] ?? [];
+            $cache       = $kirby->cache('pages');
+            $cacheId     = $this->cacheId($contentType);
+            $result      = $cache->get($cacheId);
+            $html        = $result['html'] ?? null;
+            $response    = $result['response'] ?? [];
+            $usesAuth    = $result['usesAuth'] ?? false;
+            $usesCookies = $result['usesCookies'] ?? [];
+
+            // if the request contains dynamic data that the cached response
+            // relied on, don't use the cache to allow dynamic code to run
+            if (Responder::isPrivate($usesAuth, $usesCookies) === true) {
```

Comment:
Maybe I am reading this wrong, but to me this looks like we would base the decision whether the current request uses auth or cookies on the cached values, not the current one.

Reply:
The `$usesAuth` and `$usesCookies` values come from the cache. But they only store which cookies were accessed by the previously rendered page, no matter if these cookies were already set back then. The current request is checked by `Responder::isPrivate()`. If there is an intersection between the request data and the used data, the cache is ignored.